backend/src/tools/image.py

```
@@ -0,0 +1,366 @@
+        if hasattr(message, "image_url"):
+            image_url = message.image_url
+        if hasattr(message, "image_id"):
+            image_id = message.image_id
+
+        if not image_url:
+            # Log the full response for debugging
+            logger.error(f"Failed to extract image URL from response: {response}")
+            raise ToolException(
+                "Failed to extract image URL from API response. "
+                "The image generation may have failed or the response format is unexpected."
+            )
```

Ruff (0.13.3)

Abstract `raise` to an inner function

(TRY301)